### PR TITLE
Add summary mode

### LINKED ISSUE #3494: Define SDK summary mode in QuestionnaireConfig
**Describe the feature request.**
Add a new property in the questionnaire config, to allow to use this SDK API

https://github.com/google/android-fhir/blob/6ad0a6751fea3e53d9e2be327ca205cca1112af8/datacapture/src/main/java/com/google/android/fhir/datacapture/QuestionnaireFragment.kt#L347-L351

**Additional context**
- In SDK, they call this `read-only` mode, BUT, we also already have read-only mode, PLUS the `read-only` mode from SDK looks more like summary mode (I would call it). So, to make it not confusing, we will name this `summary` mode in OpenSRP 2.
- Our `read-only` mode, is a disabled fields that contains text
- The SDK's `read-only` mode i.e. `summary` mode, is purely texts (the fields are changed to texts only)

[336119332-a874d74f-91e2-4c0c-950a-f364e333d7ea.webm](https://github.com/user-attachments/assets/a10fea14-8d24-4195-80ad-eb284d8654d4)
 
**Acceptance criteria**
- [ ] Should be able to use the read-only mode from SDK

**Area path**
A list of ordered steps in the app on usage of the feature to support anyone testing it e.g. Code reviewer, QA e.g.
1. Define a QuestionnaireConfig
2. Set the SDK summary mode via QuestionnaireConfig to true
3. Start the app
4. Open the Questionnaire

**Implementation plan (For Engineers)**
- [ ] Add a new variable for summary-mode in the QuestionnaireConfig
- [ ] Pass that variable to the QuestionnaireBuilder

### PR DESCRIPTION
**Pull request by FikriMilano**
**IMPORTANT: Where possible all PRs must be linked to a Github issue**

Fixes #3494

**Engineer Checklist**
- [ ] I have written **Unit tests** for any new feature(s) and edge cases for bug fixes
- [ ] I have added any strings visible on UI components to the `strings.xml` file
- [ ] I have updated the  [CHANGELOG.md](./CHANGELOG.md) file for any notable changes to the codebase
- [ ] I have run `./gradlew spotlessApply` and `./gradlew spotlessCheck` to check my code follows the project's style guide
- [ ] I have built and run the FHIRCore app to verify my change fixes the issue and/or does not break the app 
- [ ] I have checked that this PR does NOT introduce **breaking changes** that require an update to **_Content_** and/or **_Configs_**? _If it does add a sample here or a link to exactly what changes need to be made to the content._

**Code Reviewer Checklist**
- [ ] I have verified **Unit tests** have been written for any new feature(s) and edge cases
- [ ] I have verified any strings visible on UI components are in the `strings.xml` file
- [ ] I have verifed the [CHANGELOG.md](./CHANGELOG.md) file has any notable changes to the codebase
- [ ] I have verified the solution has been implemented in a configurable and generic way for reuseable components
- [ ] I have built and run the FHIRCore app to verify the change fixes the issue and/or does not break the app

### CODE REVIEW

File: android/quest/src/main/java/org/smartregister/fhircore/quest/ui/questionnaire/QuestionnaireActivity.kt [hunk 1]
```diff
@@ -282,6 +283,7 @@ class QuestionnaireActivity : BaseMultiLanguageActivity() {
       )
       .showAsterisk(this.questionnaireConfig.showRequiredTextAsterisk)
       .showRequiredText(this.questionnaireConfig.showRequiredText)
+      .setIsReadOnly(questionnaireConfig.isSummary())
```
Comment (ellykits) [inline android/quest/src/main/java/org/smartregister/fhircore/quest/ui/questionnaire/QuestionnaireActivity.kt:286]: LGTM. Can this be tested?

**Comment by ellykits** [comment]
Plan a demo for this @FikriMilano

**Review by ellykits — APPROVED** [review]
